metricbeat/helper/prometheus/metric.go

```diff
@@ -111,26 +133,50 @@ func LabelMetric(field, label string, options ...MetricOption) MetricMap {
 func InfoMetric(options ...MetricOption) MetricMap {
 	return &infoMetric{
 		commonMetric{
-			options: options,
+			config: Configuration{MetricProcessingOptions: options},
+		},
+	}
+}
+
+// ExtendedInfoMetric obtains info labels from the given metric and puts them
+// into events matching all the key labels present in the metric
```

Comment:
I understand the difference between this and InfoMetric is that we use it to catch all labels?

Reply:
yes, re-reading the comment that is not clear.
Will improve it, thanks

Reply:
Also, I think most existing uses of `InfoMetric` could use this extended version out of the box, so a choice was evolving current InfoMetric instead of creating the new one.

That would need a lot of testing, not in the mood for that, but I don't see a reason why we should left some labels coming from prometheus not pushed. That might be something to check for each existing use of `InfoMetrics` if this extended type gets in this PR

Reply:
I had the same thought, we should consider moving all existing metricsets to the new behavior. This could perhaps be a parameter for `InfoMetric`

Reply:
That sounds good to me, as long as we keep the plan for unifying them to avoid confusion

Reply:
https://github.com/elastic/beats/issues/14334